modules/@apostrophecms/schema/ui/apos/logic/AposInputRelationship.js

```diff
@@ -274,29 +274,19 @@ export default {
 
       const result = await apos.modal.execute(editor, {
         schema: this.field.schema,
-        item,
-        title: item.title,
-        'model-value': item._fields
+        item
       });
 
-      if (result) {
+      if (result?.image) {
         const index = this.next.findIndex(_item => _item._id === item._id);
 
         this.next = this.next.map((item, i) => {
           return i === index
-            ? {
-              ...item,
-              _fields: result
-            }
+            ? result.image
```

Comment:
I think you can refactor the `index` computation like what you did in another file in this PR

Reply:
Done